pkg/controller/versions/strategy_v2_3.go

```diff
@@ -199,6 +201,41 @@ func (v *versionStrategyV2_3) validateAddons(spec *v2.ControlPlaneSpec, allError
 	return allErrors
 }
 
+func (v *versionStrategyV2_3) validateServiceMeshExtensionsRemoved(ctx context.Context, cl client.Client, smcp metav1.Object) error {
+	logger := logf.Log.WithName("smcp-validator-2.3")
+
+	serviceMeshExtensions := &apiv1.ServiceMeshExtensionList{}
+	if err := cl.List(ctx, serviceMeshExtensions); err != nil {
+		if !errors.IsNotFound(err) && !meta.IsNoMatchError(err) {
+			// log error, but don't fail validation- this is a convenience check
```

Comment:
I think that this assumption is not correct. Let's say that a user has SME instances and upgrades to 2.3, but the operator couldn't list SMEs, because of a timeout or any other transient error. In such a case, validation will pass and upgrade will be performed while it shouldn't.

Reply:
yeah the trade-off here was between 
- accidentally blocking a valid upgrade
- accidentally allowing an upgrade where there are still SMEs in the member namespaces

I figured that accidentally blocking a valid upgrade would be worse, but I'm ready to be convinced otherwise

Reply:
My point of view is: why to validate something if we ignore errors and force passing?

Reply:
I was thinking a lot about this trade-off and my final position is that we should not allow for an upgrade in case of validation failure, because consequences of an unexpected upgrade are incomparably worse than failed upgrade that can be retried. We don't know what WASM filters customers use, so we should assume that breaking them may cause a downtime.

Reply:
You convinced me, @jewertow :wink: I made the changes, PTAL